### ISSUE TITLE
Draw pr curve with provided raw data.

### ISSUE THREAD

**Issue by foreverYoungGitHub** [lanpa/tensorboard-pytorch#98]
Hi,

Thanks so much for you to provide so convenient tools to visualize the process for pytorch. 

Recently, I want draw the pr curve and other curves with tensorboard by directly provided the precision and recall values. However, I found that the inputs of `add_pr_curve` is the predict value and ground truth. It calculated the pr curve by using compute_curve(). 

Is that possible that add a new interface like add_pr_curve_raw to add pr curve officially by provided the raw data. Just like pr_curve_raw_data_pb in tensorboard.

Thanks so much!

Best wishes,

**Comment by lanpa** [comment]
No problem.

**Comment by foreverYoungGitHub** [comment]
Thanks so much!

**Comment by lanpa** [comment]
@foreverYoungGitHub  Have a try~

**Comment by foreverYoungGitHub** [comment]
Thank you for your help!

I draw some pr_curves and other curves like this:

![pr_curve](https://raw.githubusercontent.com/ShuangXieIrene/ssds.pytorch/master/doc/imgs/pr_curve.jpg)

Thanks for your help!

Besides, do you have any plan to add something like [custom_scalar](https://github.com/tensorflow/tensorboard/tree/master/tensorboard/plugins/custom_scalar)?

Thanks again for your work!

Best wishes,

**Comment by lanpa** [comment]
So does the result make sense? `custom_scalar` is cool, I will bring it to tensorboardX.

**Comment by foreverYoungGitHub** [comment]
It is great to hear that!

The current result is great! Just some curve does not belong to PR curve and it has to normalize into 0 to 1. But it's some minor problem.

**Comment by MaximArtemev** [comment]
Hi!
Any success with bringing custom_scalar into tensorboardX?

**Comment by lanpa** [comment]
@MaximArtemev Haven't start yet. 😂

**Comment by zhanghanbin3159** [comment]
@foreverYoungGitHub how do you draw pr-curve with the precision and recall values?

**Comment by foreverYoungGitHub** [comment]
@zhanghanbin3159 please check [this code](https://github.com/ShuangXieIrene/ssds.pytorch/blob/1233516ad99c95dfb4b5cf1b29a2ced1b8351170/lib/utils/visualize_utils.py#L172)
